torch/testing/_internal/common_methods_invocations.py

```diff
@@ -15345,6 +15401,23 @@ def ref_pairwise_distance(input1, input2):
             DecorateInfo(unittest.skip("Skipped!"), "TestJit", "test_variant_consistency_jit", dtypes=(torch.float32,),),
         )
     ),
+    OpInfo(
+        "nn.functional.margin_ranking_loss",
+        ref=reference_margin_ranking_loss,
+        dtypes=floating_types_and(torch.bfloat16, torch.float16),
+        supports_out=False,
+        sample_inputs_func=sample_inputs_margin_ranking_loss,
+        skips=(
+            # False is not true : Found a sampled tensor of floating-point dtype
+            # torch.float32 sampled with requires_grad=False. If this is
+            # intended, please skip/xfail this test. Remember that sampling
+            # operations are executed under a torch.no_grad contextmanager.
+            DecorateInfo(
+                unittest.expectedFailure, "TestCommon",
+                "test_floating_inputs_are_differentiable",
+                dtypes=(torch.float32,)),
```

Comment:
This needs fixing

Reply:
so, i've looked at this again. i don't understand what i need to do here.

the error message is as follows:

>        #     # False is not true : Found a sampled tensor of floating-point dtype
>        #     # torch.float32 sampled with requires_grad=False. If this is
>        #     # intended, please skip/xfail this test. Remember that sampling
>        #     # operations are executed under a torch.no_grad contextmanager.

i don't understand this at all. so, i'm sampling with `requires_grad=False` because i'm passing `requires_grad` to kwargs. okay.

but here's a docstring of no_grad:

```
class no_grad(_DecoratorContextManager):
    r"""Context-manager that disabled gradient calculation.

    Disabling gradient calculation is useful for inference, when you are sure
    that you will not call :meth:`Tensor.backward()`. It will reduce memory
    consumption for computations that would otherwise have `requires_grad=True`.

    In this mode, the result of every computation will have
    `requires_grad=False`, even when the inputs have `requires_grad=True`.

    This context manager is thread local; it will not affect computation
    in other threads.
```

so it seems like it should be fine with `False`. it should have no effect, no?

is the error msg misleading or something?

i see that this is skipped in a bunch of places too, with a comment saying that a derivative is not implemented for some tensor. but we provide derivatives for all input tensors, no? so what's the issue here?

Reply:
You are returning a tensor with  `requires_grad=False` from your sampling function. You need to return them with `requires_grad=requires_grad`, otherwise the autograd will not be tested from them. If you do operation son them (e.g. `sign`...) they will lose the `requires_grad` attribute you created them with, as sampling functions are executed under a `no_grad` context manager.